File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let res = 0;
    let [i, j] = [0, height.length - 1];
    while (i < j) {
        const h1 = height[i];
        const h2 = height[j];
        const area = Math.min(h1, h2) * (j - i);
        res = Math.max(res, area);
        if (h1 < h2) {
            i++;
        } else j--;
    }
    return res;
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        res = 0
        i = 0
        j = len(height) - 1

        while i < j:
            h1 = height[i]
            h2 = height[j]
            area = min(h1, h2) * (j - i)
            res = max(area, res)
            if h1 < h2:
                i += 1
            else:
                j -= 1
        return res